asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
+that determine the semantic relation they pertain to, while output configurations
+use modes to distinguish between conceptually different kinds of outputs, for example
+ones where an exception was raised, ones when a dynamic error occurred, etc.
+
+We now explain the components over which configurations are defines.
+
+\subsection{Native Values}
+The semantics of an ASL specification associates \emph{native values} to variables.
+The set of native values is the minimal set defined by the following recursive rules (\texttt{NV} stands for Native Value):
+\begin{mathpar}
+  \inferrule[(Basis Set: Integers, Reals, Booleans, Strings, and Bitvectors)]{\vv \in \literal}
+  { \nvliteral{v} \in \vals }
+  \and
+  \inferrule[(Tuple Values and Array Values)]{i=1..k: v_i \in \vals}
+  { \nvvector{v_{1..k}} \in \vals }
+  \and
+  \inferrule[(Record Values)]{m \in \identifier \rightarrow_{\text{fin}} \vals}
+  { \nvrecord{m} \in \vals }
+\end{mathpar}
+(The notation $\rightarrow_{\text{fin}}$ stands for a function over a finite domain.)
+
+\newcommand\nvint[0]{\texttt{Int}}
+\newcommand\nvbool[0]{\texttt{Bool}}
+\newcommand\nvreal[0]{\texttt{Real}}
+\newcommand\nvstring[0]{\texttt{String}}
+\newcommand\nvbitvector[0]{\texttt{Bitvector}}
+We use the following shorthands for native value literals:
+\[
+\begin{array}{rcl}
+\nvint(z)       &\triangleq& \nvliteral{\lint(z)}\\
+\nvbool(b)      &\triangleq& \nvliteral{\lbool(b)}\\
+\nvreal(r)      &\triangleq& \nvliteral{\lreal(r)}\\
+\nvstring(s)    &\triangleq& \nvliteral{\lstring(s)}\\
+\nvbitvector(v) &\triangleq& \nvliteral{\lbitvector(v)}\\
+\end{array}
+\]
+
+\newcommand\tint[0]{\mathcal{Z}}
+\newcommand\tbool[0]{\mathcal{B}}
+\newcommand\treal[0]{\mathcal{R}}
+\newcommand\tstring[0]{\mathcal{S}\mathcal{T}\mathcal{R}}
+\newcommand\tbitvector[0]{\mathcal{B}\mathcal{V}}
+\newcommand\tvector[0]{\mathcal{V}\mathcal{E}\mathcal{C}}
+\newcommand\trecord[0]{\mathcal{R}\mathcal{E}\mathcal{C}}
+We use the following shorthands for different types of native values:
+\[
+\begin{array}{rcl}
+\tint       &\triangleq& \{ \nvint(z) \;|\; z \in \Z\}\\
+\tbool      &\triangleq& \{ \nvbool(\True), \nvbool(\False) \}\\
+\treal      &\triangleq& \{ \nvreal(r) \;|\; r \in \Q\}\\
+\tstring    &\triangleq& \{ \nvstring(s) \;|\; \texttt{"}s\texttt{"} \in \texttt{<string>}\}\\
+\tbitvector &\triangleq& \{ \nvbitvector(\textit{bits}) \;|\; \textit{bits} \in \{0,1\}^+\}\\
+\tvector    &\triangleq& \{ \nvvector{\textit{vals}} \;|\; \textit{vals} \in \vals^*\}\\
+\trecord    &\triangleq& \{ \nvrecord{\fieldmap} \;|\; \fieldmap \in \identifier\rightarrow\vals\}\\
+\end{array}
+\]
+
+\subsection{Dynamic Environments}
+
+A \emph{sequential dynamic environment}, or \emph{dynamic environment}, for short, is a structure which,
+amongst other things, associates native values to variables.
+Formally, a sequential environment $\denv \in \dynamicenvs$ is a pair consisting of a partial function from global
```

Comment:
please define "partial function"

Reply:
This is defined in "Functions and Products".

Reply:
please add a pointer to it.